src/Extension/FluentExtension.php

```diff
@@ -679,6 +685,86 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
                 SQLUpdate::create($localisedTable, $fieldsToUpdate, ['RecordID' => $toID])->execute();
             }
         }
+
+        // We don't have any localised relations to cover
+        if (count($copyHasOneRelations) === 0) {
+            return;
+        }
+
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        // We don't have a current locale which indicates that the Fluent setup is incomplete - bail out
+        if (!$currentLocale) {
+            return;
+        }
+
+        $locales = [];
+
+        /** @var RecordLocale $localeInformation */
+        foreach ($owner->Locales() as $localeInformation) {
+            $sourceLocale = $localeInformation->getSourceLocale();
+            $modelLocale = $localeInformation->getLocaleObject();
+
+            if (!$sourceLocale) {
+                // We don't have any source locale, so we can bail out
+                continue;
+            }
+
+            if ($modelLocale->Locale !== $sourceLocale->Locale) {
+                // Source of this locale is different from current locale, so we can skip it
+                // as this locale content is being inherited
+                continue;
+            }
+
+            if ($modelLocale->Locale === $currentLocale) {
+                // Current locale can be skipped as it was already handled correctly
+                continue;
+            }
+
+            // Add locale which uses current locale as a source to our list
+            $locales[] = $modelLocale->Locale;
+        }
```

Comment:
This feels like it should probably be its own method

Reply:
Yes, agree  - particularly it would be useful to extract out a separate method that gives you a list of locale codes that the model is localised in for a particular stage as we already use a similar feature in multiple projects. I'm happy to make these changes but I don't think I can make it in this PR and stay within patch version contraints.

Reply:
Making a private method is perfectly fine in a patch.
Its visibility could then be changed later in a minor release if it seems like it could be useful for purposes outside of this specific class.